chalice/config.py

```diff
@@ -231,13 +242,35 @@ def manage_iam_role(self):
     def autogen_policy(self):
         # type: () -> bool
         return self._chain_lookup('autogen_policy',
-                                  varies_per_chalice_stage=True)
+                                  varies_per_chalice_stage=True,
+                                  varies_per_function=True)
 
     @property
     def environment_variables(self):
         # type: () -> Dict[str, str]
         return self._chain_merge('environment_variables')
 
+    @property
+    def tags(self):
+        # type: () -> Dict[str, str]
+        tags = self._chain_merge('tags')
+        tags['aws-chalice'] = 'version=%s:stage=%s:app=%s' % (
+            current_chalice_version, self.chalice_stage, self.app_name)
+        return tags
+
+    def scope(self, chalice_stage, function_name):
+        # type: (str, str) -> Config
+        # Used to create a new config object that's scoped to a different
+        # stage and/or function.  This creates a completely separate copy.
```

Comment:
This API is actually wrong now.  Initially I was doing a `deepcopy` on the `self.__dict__`, but ran into issues that couldn't be serialized because our config obj has a reference to the `chalice_app` which contains all contains of objects.  Now that we're sharing `__dict__`, this doesn't actually work.  I'll get this fixed.